### DISCUSSION TITLE
"Can't find root node." if `<div id="remark42"></div>` is missing

### DISCUSSION THREAD

**Discussion by pat-s** [umputun/remark42#1732] [General]
Related to https://github.com/umputun/remark42/discussions/1707

Remarks throws a console error after installation on every page where the `<div id="remark42"></div>` is missing. It doesn't matter whether it's used on other pages below the website root.

Usually the snippet is only placed into blog posts and hence the above will always throw the error on the website root. This causes penalties in website scores. Is there a way to avoid this?

**Comment by paskal** [comment]
Sure, don't include remark42 initialisation on the page or adjust the `components` list [according to the doc](https://remark42.com/docs/configuration/frontend/): it should be equal to `['counter']` on the page where you don't expect to show comments but have only list of posts and counters for amount of comments under them.

**Reply by pat-s** [reply]
Thanks. I have the config and components init script in `head.html` which means it's included on every page. This means it also searches for the above mentioned `div` on every page, otherwise the warning is shown.

I now found out that I need to place the following on each individual site where I want a widget to show up

```
<script>
  !(function (e, n) {
    for (var o = 0; o < e.length; o++) {
      var r = n.createElement("script"),
        c = ".js",
        d = n.head || n.body;
      "noModule" in r ? ((r.type = "module"), (c = ".mjs")) : (r.async = !0),
        (r.defer = !0),
        (r.src = remark_config.host + "/web/" + e[o] + c),
        d.appendChild(r);
    }
  })(remark_config.components || ["embed"], document);
</script>

<div id="remark42"></div>
```

while the overall config can stay in `head.html`.

```
  <script>
    var remark_config = {
    };
  </script>
```

Is this the correct approach?

**Reply by paskal** [reply]
I think it's the correct approach. Please see [this search](https://github.com/search?q=repo%3Axianmin%2Fhugo-theme-jane%20remark42&type=code) for https://github.com/xianmin/hugo-theme-jane integration I did, it loads the minimal necessary amount of stuff needed for comments and only on relevant pages.

**Reply by avinal** [reply]
Thank you, that helped
